### ISSUE TITLE
Built-in browser

### ISSUE THREAD

**Issue by goerz** [blinksh/blink#143]
This is probably still a bit far out there, in terms of priorities for the next few releases, but in the long term, it would be really nice to have a built-in browser. My particular use case is the [Jupyter Notebook](http://jupyter.org). This goes very much hand-in-hand with #142, since the notebook server I'd be connecting to is only reachable via SSH tunneling (this is a very common setup).

My recommended UI for this would be a "browse" command, as in `browse https://localhost:8888` that would open the URL in a new tab (with terminal and browser tabs being intermixed, so that I can swipe left and right to switch between them).

Having a terminal, port forwarding, and the notebook all in the same app would be an absolute powerhouse, and would allow to do some very real science on the iPad (this is me talking as a physicist, but I think data science and machine learning uses very similar workflows, and everybody and their dog is doing that these days). Web devs would probably also use this feature.

It would put Blink very far ahead of the competition. [Coda](https://panic.com/coda/) in particular already has the combination of terminal and browser, but no port forwarding. That means that I need to run another application ([vSSH](http://www.velestar.com/Pages/VSSHIOSPage.aspx)) in the background just for the port forwarding, and switch back to it every 10 minutes so that iOS won't kill the connection - quite annoying. Blink also has a far cleaner and snappier interface than Coda at this time, as well as better font rendering.

**Comment by carloscabanero** [comment]
This is actually easier than it looks like, and it was one of the reasons why we chose a browser based terminal. 

Would you be ok, switching spaces all the time between the terminal and browser? Extra points for splitting the screen between the Shell and the new browser?

**Comment by goerz** [comment]
That's awesome. Personally, I'm more of a full-screen kind of guy, so I rarely split the screen at all. So switching would generally be sufficient for me (especially because if I actually want to have a terminal and a browser next to each other, I can use the general iOS split feature, and for terminals, there's tmux).

Other people might like it, though, so I wouldn't complain about having the option.

**Comment by kirillseva** [comment]
ipad pro has some neat multitasking, so opening a safari window next to blink+vim does not sound bad at all! Just need to have either port forwarding enabled. Alternatively one can host webapp/jupyter on some port accessible to www and it will *just work*

**Comment by goerz** [comment]
Yes, for a split view the iOS multitasking is perfectly sufficient. The only situation where a builtin-browser would be useful is if (a) I can only access the webapp via port forwarding and (b) I want to  work in the webapp for extended periods of time in full screen. In this situation, multitasking doesn't work, because iOS will kill the port forwarding; the app providing the port forwarding and showing the webapp have to be the same (Blink).

In my situation, both conditions are fulfilled, (a) because the compute server is firewalled (and it's a unique machine providing HPC resources, GPUs, etc, not a throwaway server I can just spin up in the cloud), and (b) because the jupyter notebook requires a certain minimum screen size for the UI not to break (i.e., it really only works well in full screen mode on an iPad Pro)

**Comment by BinaryBen** [comment]
A browser in Blink would mean I can switch between browser and terminal a lot easier using keyboard shortcuts. If split screen is added, then this could mean split screen for terminals as well? That would add extra points in my eyes

**Comment by richardanaya** [comment]
I know this may sound weird, but I program on my iPhone, and there's none of this fancy app switching functionality there. Having a built in browser would help me out a lot.

**Comment by dseuss** [comment]
Also, it would be nice to keep the remapping features in the browser

**Comment by willpuckett** [comment]
You've probably already found this, but Juno Connect (https://juno.sh/junoconnect/) has SSH port forwarding and displays Jupyter notebooks beautifully.

**Comment by DAddYE** [comment]
> This is actually easier than it looks like, and it was one of the reasons why we chose a browser based terminal.
> 
> Would you be ok, switching spaces all the time between the terminal and browser? Extra points for splitting the screen between the Shell and the new browser?

Hi Carlos,

If you have any pointer for me to ramp-up the development of this feature, I'll be happy to submit a PR.

**Comment by goerz** [comment]
Wasn't there [this recent tweet](https://twitter.com/BlinkShell/status/1263836177356017664)? Is this something that will show up soon in Testflight?
